kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/ModalContainer.kt

```diff
@@ -261,20 +261,20 @@ private class OnDestroy(private val block: () -> Unit) : LifecycleObserver {
   fun onDestroy() = block()
 }
 
-private val Dialog.activity: AppCompatActivity?
-  get() = decorView?.context?.activity()
+private val Dialog.lifecycle: Lifecycle?
+  get() = decorView?.context?.lifecycle()
 
 private val Dialog.decorView: View?
   get() = window?.decorView
 
 /**
- * The [AppCompatActivity] for this context, or null if there isn't one.
+ * The [Lifecycle] for this context, or null if one can't be found.
  *
  * We keep all this very forgiving because we're just using it to keep some logging
- * noise out of logcat. If someone manages to run this under the wrong type of
- * activity, just return null and let the caller no-op.
+ * noise out of logcat. If someone manages to run this under a strange context whose
+ * [Lifecycle] we can't find, just return null and let the caller no-op.
  */
-tailrec fun Context.activity(): AppCompatActivity? = when {
-  this is Activity -> this as? AppCompatActivity
-  else -> (this as? ContextWrapper)?.baseContext?.activity()
+tailrec fun Context.lifecycle(): Lifecycle? = when {
```

Comment:
I'm really surprised these don't come out of the box… seems like it would be a really common use case?